src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -95,34 +95,32 @@ ArrayBase::~ArrayBase() {}
  * If there was already a buffer allocated, it will be released.
  * The buffer will be deleted when the last copy of this class has been deleted.
  */
-void ArrayBase::allocateBuffer(size_t count) {
+char *ArrayBase::allocateBuffer(size_t count) {
   // Note that you can allocate a buffer of size zero.
   // The C++ spec (5.3.4/7) requires such a new request to return
   // a non-NULL value which is safe to delete.  This allows us to
   // disambiguate uninitialized ArrayBases and ArrayBases initialized with
   // size zero.
   count_ = count;
-  capacity_ = count_ * BasicType::getSize(type_);
   if (type_ == NTA_BasicType_SDR) {
     std::vector<UInt> dimension;
     dimension.push_back((UInt)count);
     allocateBuffer(dimension);
   } else {
-    std::shared_ptr<char> sp(new char[capacity_],
+    std::shared_ptr<char> sp(new char[count_ * BasicType::getSize(type_)],
                              std::default_delete<char[]>());
     buffer_ = sp;
-    own_ = true;
```

Comment:
nice to see these "owns buffer" go away

Reply:
> nice to see these "owns buffer" go away

That was for when Array was also handling external buffers and Sparse data which had variable length.  I am just now finally getting around to removing some of the leftover stuff.